src/lib/npm.ts

```diff
@@ -130,90 +128,202 @@ export async function runNpm(args: Omit<ExecPromiseArg, "command">) {
+        packageJSONPath,
+      )} and running install command again`,
     );
+  }
 
-    await runNpm({
-      args: ["i"],
-      options: {
-        cwd: await ensureFluenceAquaDependenciesPath(),
-      },
-    });
+  fluenceConfig.aquaDependencies = aquaDependencies;
+  await fluenceConfig.$commit();
 
-    return;
+  commandObj.logToStderr(
+    `${packageNameAndVersion} aqua dependency is successfully installed`,
+  );
+}
+
+async function ensureNpmInstallArgIsCorrect(packageNameAndVersion: string) {
+  const filePath = getFilePath(packageNameAndVersion);
+  const fluenceAquaDependenciesPath = await ensureFluenceAquaDependenciesPath();
+
+  if (filePath === undefined) {
+    const packageNameAndVersionTuple = splitPackageNameAndVersion(
+      packageNameAndVersion,
+    );
+
+    let [, versionToInstall] = packageNameAndVersionTuple;
+
+    if (versionToInstall !== undefined) {
+      versionToInstall = ensurePathInVersionIsCorrect(
+        versionToInstall,
+        fluenceAquaDependenciesPath,
+      );
+    }
+
+    return [packageNameAndVersionTuple[0], versionToInstall]
+      .filter(Boolean)
+      .join("@");
   }
 
-  const packageNameAndVersionTuple = splitPackageNameAndVersion(
+  return ensurePathInVersionIsCorrect(
     packageNameAndVersion,
+    fluenceAquaDependenciesPath,
+  );
+}
+
+async function updatePackageJSON(fluenceConfig: FluenceConfig) {
+  const fluenceAquaDependenciesPath = await ensureFluenceAquaDependenciesPath();
+
+  const dependenciesWithFixedRelativePath = Object.fromEntries(
+    Object.entries(fluenceConfig.aquaDependencies).map(
+      ([packageName, version]) => {
+        return [
+          packageName,
+          ensurePathInVersionIsCorrect(version, fluenceAquaDependenciesPath),
+        ];
+      },
+    ),
   );
 
-  const [packageName] = packageNameAndVersionTuple;
-  let [, version] = packageNameAndVersionTuple;
+  await writeFile(
+    await getFluenceAquaDependenciesPackageJsonPath(),
+    jsonStringify({ dependencies: dependenciesWithFixedRelativePath }),
+    FS_OPTIONS,
+  );
+}
 
-  if (version === undefined) {
-    version = await getLatestVersionOfNPMDependency(packageName);
+async function getDependenciesFromPackageJSON(parsedPackageJSON: unknown) {
+  if (
+    !hasKey("dependencies", parsedPackageJSON) ||
+    typeof parsedPackageJSON.dependencies !== "object" ||
+    parsedPackageJSON.dependencies === null
+  ) {
+    return null;
   }
 
-  await runNpm({
-    args: ["i", packageNameAndVersion],
-    options: {
-      cwd: await ensureFluenceAquaDependenciesPath(),
-    },
-  });
+  const dependenciesWithVersions = Object.entries(
+    parsedPackageJSON.dependencies,
+  );
 
-  fluenceConfig.dependencies.npm[packageName] = version;
-  await fluenceConfig.$commit();
+  if (
+    !dependenciesWithVersions.every((a): a is [string, string] => {
+      return typeof a[1] === "string";
+    })
+  ) {
+    return null;
+  }
```

Comment:
You do a lot of manual validation here.
It's great that you validate it, but the code looks too verbose and not easy to read. I think it would be convenient and concise if we would have some kind of validation schema.
I suggest to use the schemas everywhere except very simple cases at it gives better readability and nearly instantly explains what's the code about.